System/Console/ANSI/Color.hs

```diff
@@ -0,0 +1,102 @@
+-- | Conversions between color representations
+module System.Console.ANSI.Color
+  (
+  -- | 8-bit to 24-bit color conversion
+    color8To24
+  -- | Utility functions
+  , clamp
+  , color8CodeToEither
+  , closest4bitsANSIColor
+    ) where
+
+
+import qualified Data.Colour.SRGB.Linear as SRGBLin(rgb, Colour)
+import qualified Data.Colour.SRGB as SRGB (RGB (..), toSRGB, Colour)
+
+import Data.Colour.Names (black, blue, cyan, green, grey, lime, magenta, maroon,
+                          navy, olive, purple, red, silver, teal, white, yellow)
+
+import Data.List (minimumBy, (!!))
+
+import System.Console.ANSI.Types
+
+
+color8To24 :: Color8 -> SRGBLin.Colour Float
+color8To24 (RGB8Color r' g' b') =
+  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapRGB8bitComponent . (\v -> clamp v 0 5) :: Int -> Float
+      r = mapping r'
+      g = mapping g'
+      b = mapping b'
+  in  SRGBLin.rgb r g b
+color8To24 (Gray8Color g') =
+  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapGray8bitComponent . (\v -> clamp v 0 23) :: Int -> Float
+      g = mapping g'
+  in  SRGBLin.rgb g g g
+
+-- | how xterm interprets 8bit rgb colors (deduced from https://jonasjacek.github.io/colors/)
+{-# INLINE xtermMapRGB8bitComponent #-}
```

Comment:
Are these INLINE pragmas needed?

Reply:
I was under the impression that these functions are "small enough" so that inlining them would make the program performance better.

Reply:
They are small enough, and ghc is probably smart enough to make them inlinable. And if ghc doesn't inline them, that perhaps it has a good reason to. So I wouldn't add INLINE pragmas unless there's  a good explicit reason to do so.

Reply:
ok, I removed them.